### PR TITLE
Improve/fix modal navigation

### PR DESCRIPTION
**Pull request by benjaminpkane**
## What changes are proposed in this pull request?

Fixes modal navigation during prolonged use

## How is this patch tested? If it is not, please explain why.

Navigating all 200 `quickstart` samples from start to finish to start.

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Introduced a new user prompt message: "slow down, you're clicking a lot" to enhance user experience.
	- Added a new class for improved data iteration through sections.
	- Enhanced label rendering capabilities with support for temporal detection.

- **Bug Fixes**
	- Improved error handling in the Section class to prevent repeated requests when conditions are not met.

- **Refactor**
	- Streamlined control flow and conditional checks in the Section class for enhanced readability and robustness.
	- Simplified the `createScrollReader` functionality to focus on essential scroll state management.

- **Improvements**
	- Enhanced navigation capabilities with new methods for iterating through sections.
	- Adjusted default grid zoom range to allow for a broader range of zoom levels.
	- Improved cleanup logic in the Grid component for better resource management.

### CODE REVIEW

File: app/packages/spotlight/src/section.ts [hunk 1]
```diff
@@ -238,17 +245,64 @@ export class Section<K, V> {
...
+
+    if (this.#direction === DIRECTION.FORWARD) {
+      await this.next(request, renderer, sibling);
+      if (this.#direction === DIRECTION.FORWARD) {
+        return this.#nextMap.get(id);
+      }
+    }
+
+    throw await sibling(false);
+  }
```
Comment (sashankaryal) [inline app/packages/spotlight/src/section.ts:267]: 🥇

File: app/packages/core/src/components/Grid/Grid.tsx [hunk 2]
```diff
@@ -114,6 +115,7 @@ function Grid() {
     spotlight.addEventListener("rowchange", set);
 
     return () => {
+      freeVideos();
```
Comment (sashankaryal) [inline app/packages/core/src/components/Grid/Grid.tsx:118]: 😮

**Review by sashankaryal — APPROVED** [review]
I like the addition of the `Iter` class, neat!
It's a little hard to grok the relationship / ownership between different components - I guess an opportunity to improve spotlight's readme in the future.

File: app/packages/spotlight/src/section.ts (hunk 1 above)
Comment (benjaminpkane) [reply app/packages/spotlight/src/section.ts:267]: API needs better naming 😓

**Comment by benjaminpkane** [comment]
> I like the addition of the `Iter` class, neat! It's a little hard to grok the relationship / ownership between different components - I guess an opportunity to improve spotlight's readme in the future.

`Iter` class worked out well, but is still a little disorganized. I can't be sure, but I believe this way `Section` instances will still be garbage collected because iteration is now scoped in a separate class. Would like to cleanup / move it out of `Section` and into `Spotlight` so only one instance is created per spotlight instance.

**Pull request merged**